checker/src/org/checkerframework/checker/lock/LockTransfer.java

```diff
@@ -173,4 +196,18 @@ else if (astKind == Kind.ARBITRARY_CODE) { // Handle initializers
 
         return result;
     }
+
+    @Override
+    public TransferResult<CFValue, LockStore> visitStringConcatenate(
```

Comment:
I can't remember if this will be called for a `StringConcatenateAssignmentNode`, too. (Or if the LockTreeAnnotator will take care of that.) You should add a test and if it fails then you'll want to override `visitStringConcatenateAssignmentNode` and add the same behavior.